master/buildbot/process/buildrequestdistributor.py

```diff
@@ -372,15 +372,20 @@ def _defaultSorter(self, master, builders):
 
         @defer.inlineCallbacks
         def key(bldr):
-            # Sort by time of oldest build request
+            # Sort primarily highest priority of build requests
+            priority = yield bldr.getHighestPriority()
+            if priority is None:
+                # for builders that do not have pending buildrequest, we just use large number
+                priority = -math.inf
+            # Break ties using the time of oldest build request
```

Comment:
The first commit sorts builds within a single builder. This one sorts the builders.

Neither of my sort changes would change default behavior, since by default all requests have priority 0 and python's sort is stable. Maybe I could make this explicit in the commit messages.

But it seems counterintuitive to me that there is a priority field if it is completely ignored with the default settings.

Reply:
Ah. And I guess we can't affect the builder chosen using nextBuild... Lets hold off on this portion.

I would say that buildbot doesn't want to set the policy of what the build requests should be sorted by. That's why they provide the nextBuild() method so the user can sort them. It's an open question to how to chose which builder has priority.

Lets leave that part for a potential third PR then. If it doesn't change behaviour than it should probably be ok.

Reply:
> Ah. And I guess we can't affect the builder chosen using nextBuild... Lets hold off on this portion.

Yup. There is a builder priority function that can be customized: http://docs.buildbot.net/current/manual/customization.html#builder-priority-functions

> Lets leave that part for a potential third PR then. If it doesn't change behaviour than it should probably be ok.

Works for me. I'll mention when I open the first PR that there is more coming down the line

Reply:
Ok,great. Then I think it makes sense to leave the default order as simple as possible. We can set those functions in eid-buildbot just fine.